jupyterhub/app.py

```diff
@@ -704,6 +704,21 @@ def _bind_url_default(self):
         proto = 'https' if self.ssl_cert else 'http'
         return proto + '://:8000'
 
+    public_url = Unicode(
+        "",
+        config=True,
+        help="""Set the public URL of JupyterHub
+
+        This will skip any detection of URL and protocol from requests,
+        which isn't always correct when JupyterHub is behind
+        multiple layers of proxies, etc.
+        Usually the failure is detecting http when it's really https.
+
+        Should include the full, public URL of JupyterHub,
+        including the public-facing base_url prefix.
```

Comment:
Can you specify if this should or should not include a trailing slash?

Reply:
Indeed it should, and we can add a validator to _ensure_ it does, so that common mistakes can't cause problems.

Reply:
+1